packages/block-library/src/block/edit.js

```diff
@@ -401,47 +215,16 @@ function ReusableBlockEdit( {
 		),
 	} );
 
+	// Use `blocks` variable until `innerBlocks` is populated, which has the proper clientIds.
 	const innerBlocksProps = useInnerBlocksProps( blockProps, {
 		templateLock: 'contentOnly',
 		layout,
-		renderAppender: innerBlocks?.length
-			? undefined
-			: InnerBlocks.ButtonBlockAppender,
+		value: innerBlocks.length > 0 ? innerBlocks : blocks,
```

Comment:
@ellatrix, @SantosGuillamot, can you provide additional context around this change?

> Use `blocks` variable until `innerBlocks` is populated, which has the proper clientIds.

What does proper clientIds mean here?

This seems to be the cause of the infinite loop regression in #63909.

Reply:
It's possibly related to the way `useBlockSync` clones the inner blocks for controlled blocks:
https://github.com/WordPress/gutenberg/blob/54eee32b043ff1879cba2f82f9d5568df43bc34c/packages/block-editor/src/components/provider/use-block-sync.js#L114

That gives the blocks new client ids.

Having different client ids is important. If you insert two instances of the same pattern into a post the inner blocks need unique client ids so that updates to one don't update the other.

Reply:
Sorry, I've been away for a few days 🙂 

I remember having issues with having different `clientIds` in `blocks` and in `innerBlocks` as explained in [this comment](https://github.com/WordPress/gutenberg/pull/60721#issuecomment-2077765846). I even refer to these lines there. It seems the `innerBlocks` clientIds were actually the ones in the markup, that's what I was trying to refer with "proper" clientIds.

What I don't remember is if I added that check because it was breaking something or just because I thought it was correct.

I'm fine trying to remove it and see if the tests pass.